server/routes/journeys/develop-an-initial-plan/check-answers/controller.ts

```diff
@@ -0,0 +1,61 @@
+import { NextFunction, Request, Response } from 'express'
+import { BaseJourneyController } from '../../base/controller'
+import { SanitisedError } from '../../../../sanitisedError'
+
+export class PlanCheckAnswersController extends BaseJourneyController {
+  GET = async (req: Request, res: Response) => {
+    req.journeyData.isCheckAnswers = true
+
+    const { plan } = req.journeyData
+    const sortedNeeds = plan!.identifiedNeeds?.sort((needA, needB) => {
+      return new Date(needA.createdDate).getTime() - new Date(needB.createdDate).getTime()
+    })
+    res.render('develop-an-initial-plan/check-answers/view', {
+      plan: {
+        ...plan,
+        identifiedNeeds: sortedNeeds,
+      },
+      csipRecordUrl: `/csip-records/${req.journeyData.csipRecord!.recordUuid}`,
+    })
+  }
+
+  checkSubmitToAPI = async (req: Request, res: Response, next: NextFunction) => {
+    const plan = req.journeyData.plan!
+    try {
+      await this.csipApiService.createPlan(req, {
+        caseManager: plan.caseManager!,
+        firstCaseReviewDate: plan.firstCaseReviewDate!,
+        reasonForPlan: plan.reasonForPlan!,
+        identifiedNeeds: plan.identifiedNeeds!.map(
+          ({ closedDate, createdDate, identifiedNeed, intervention, progression, responsiblePerson, targetDate }) => ({
+            createdDate,
+            identifiedNeed,
+            intervention,
+            responsiblePerson,
+            targetDate,
+            closedDate: closedDate!,
+            progression: progression!,
```

Comment:
`closedDate` is quite definitely `null` when we develop an initial plan. i don't think we have a question screen for it.
`progression` is optional question, so it can be null and can't be `!`'d here.

Reply:
`createdDate` is to be auto populated with today's date.
i missed setting this value in the PR for `/develop-an-initial-plan/record-actions-progress`.
i think we either add the value over there or here.
probably better to set the date when we push the subJourney identified need into the identifiedNeeds array.

Reply:
Done on both counts